### PR TITLE
Add integration test for `RedisStorage`

### PR DESCRIPTION
**Pull request by masap**
## Motivation
Add integration test for `RedisStorage`

### CODE REVIEW

**Comment by HideakiImamura** [comment]
The redis URL cannot be passed to RDBStorage. We need to instantiate the RedisStorage.

**Comment by masap** [comment]
PR #3238 breaks it. I will fix.

**Comment by masap** [comment]
I already fixed it.

**Comment by masap** [comment]
Fixed.

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [inline tests/storages_tests/test_with_server.py:124]: I think quite a few users want to get raw storage because it significantly increases the number of queries. So, I'm a bit conservative to introduce the `raw` option to `get_storage`.

How about getting the raw storage by accessing the `storage._backend` property?

```python
cached_storage = optuna.storages.get_storage(storage_url)
assert isinstance(cached_storage, _CachedStorage)
storage: Union[RDBStorage, RedisStorage] = cached_storage._backend
```

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [inline tests/storages_tests/test_with_server.py:128]: If we have a tracking issue about supporting `float(inf)`, could you add a link for the developers? In addition, how about replacing this silent return with `pytest.skip("RedisStorage does not accept float('inf')")` ?

File: .github/workflows/tests-dbstorage.yml
Comment (toshihikoyanase) [inline .github/workflows/tests-dbstorage.yml:1]: Could you update the job name of `test-rdbstorage` in L16 as well? I guess `test-storage-with-server` might explain the contents of the job.

File: .github/workflows/tests-dbstorage.yml
Comment (toshihikoyanase) [inline .github/workflows/tests-dbstorage.yml:68]: Since Optuna doesn't have `DBStorage`, how about putting emphasis on `test with server`? 
```suggestion
        cache-name: test-storage-with-server
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR!
I'm not aware of the most of the test cases work with the Redis server without any changes.

I have some comments on the job name and the conditional branch in the test case. In addition to that, I'm thinking of the name of the workflow file can be `tests-storage-with-server.yml`. What do you think of it?
cc. @HideakiImamura

File: tests/storages_tests/test_with_server.py
Comment (masap) [reply tests/storages_tests/test_with_server.py:124]: > I think quite a few users want to get raw storage because it significantly
> increases the number of queries. So, I'm a bit conservative to introduce
> the raw option to get_storage.

I've heard that there is a plan to add a raw option to get_storage().
https://github.com/optuna/optuna/pull/3204#discussion_r789365501

Therefore, I have already added this comment.
https://github.com/masap/optuna/blob/a397bb1835754cbc2d510764b12a0a0ca4839762/tests/storages_tests/test_storages.py#L1251-L1255

Could you form a consensus on the implementation policy of get_storage() in optuna maintainer please?

File: .github/workflows/tests-dbstorage.yml
Comment (masap) [reply .github/workflows/tests-dbstorage.yml:1]: I already mentioned about it at the first commit log, "I left the job name as `tests-rdbstorage` because I do not have permission to edit `Branch protection rules`". When we change the job name, we should also change the `Branch protection rules`. Without it, CI will fail.

So could you please modify it after this PR is merged?

File: tests/storages_tests/test_with_server.py
Comment (HideakiImamura) [reply tests/storages_tests/test_with_server.py:128]: I have confirmed that the current Redis storage already supports `inf` to store it in the storage. 

<details>
<summary>Test for study.optimize</summary>

```python
import optuna
import numpy as np

def objective(trial):
    return float(np.inf)

# In-memory storage
study = optuna.create_study()
study.optimize(objective, n_trials=1)
print(f"In-memory : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (SQLite)
study = optuna.create_study(storage='sqlite:///../sqlite.db')
study.optimize(objective, n_trials=1)
print(f"SQLite    : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (MySQL)
study = optuna.create_study(storage="mysql+pymysql://root:test@localhost/mysql")
try:
    study.optimize(objective, n_trials=1)
except Exception as e:
    print(e)
print(f"MySQL     : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (PostgreSQL)
study = optuna.create_study(storage="postgresql+psycopg2://postgres:test@localhost:15432/postgres")
study.optimize(objective, n_trials=1)
print(f"PostgreSQL: Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# Redis storage
study = optuna.create_study(storage="redis://localhost:6379/0")
try:
    study.optimize(objective, n_trials=1)
except Exception as e:
    print(e)
finally:
    print(f"Redis     : Trial state is {study.trials[-1].state}")
print(study.trials[-1])
```

```
(venv) mamu@HideakinoMacBook-puro optuna % python inf-objective.py
[I 2022-02-03 16:38:09,480] A new study created in memory with name: no-name-5b41f645-2d08-4af3-b343-b8cfd447e88e
[I 2022-02-03 16:38:09,481] Trial 0 finished with value: inf and parameters: {}. Best is trial 0 with value: inf.
In-memory : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 38, 9, 481592), datetime_complete=datetime.datetime(2022, 2, 3, 16, 38, 9, 481728), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=0, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:38:09,548] A new study created in RDB with name: no-name-d3c8c30d-ed7e-4d8e-82b9-fa0ddd75127f
[I 2022-02-03 16:38:09,611] Trial 0 finished with value: inf and parameters: {}. Best is trial 0 with value: inf.
SQLite    : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 38, 9, 569079), datetime_complete=datetime.datetime(2022, 2, 3, 16, 38, 9, 582011), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=24, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:38:09,742] A new study created in RDB with name: no-name-f8300a3d-90c1-4909-bf2a-3c7ec1042da1
[I 2022-02-03 16:38:09,940] Trial 0 finished with value: inf and parameters: {}. Best is trial 0 with value: inf.
MySQL     : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 38, 9, 793344), datetime_complete=datetime.datetime(2022, 2, 3, 16, 38, 10), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=48, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:38:10,171] A new study created in RDB with name: no-name-eae39a96-126f-4af9-a43e-5f4d1def5146
[I 2022-02-03 16:38:10,317] Trial 0 finished with value: inf and parameters: {}. Best is trial 0 with value: inf.
PostgreSQL: Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 38, 10, 215340), datetime_complete=datetime.datetime(2022, 2, 3, 16, 38, 10, 245949), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=27, state=TrialState.COMPLETE, value=None)
/Users/mamu/Dev/pfn/optuna/optuna/storages/__init__.py:29: ExperimentalWarning: RedisStorage is experimental (supported from v1.4.0). The interface can change in the future.
  return _CachedStorage(RedisStorage(storage))
[I 2022-02-03 16:38:10,336] A new study created in Redis with name: no-name-0000000016
[I 2022-02-03 16:38:10,343] Trial 0 finished with value: inf and parameters: {}. Best is trial 0 with value: inf.
Redis     : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 38, 10, 338778), datetime_complete=datetime.datetime(2022, 2, 3, 16, 38, 10, 340677), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=16, state=TrialState.COMPLETE, value=None)
```

</details>

<details>
<summary>Test for trial.report</summary>

```python
import optuna
import numpy as np

def objective(trial):
    trial.report(float(np.inf), 1)
    return 1

# In-memory storage
study = optuna.create_study()
study.optimize(objective, n_trials=1)
print(f"In-memory : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (SQLite)
study = optuna.create_study(storage='sqlite:///../sqlite.db')
study.optimize(objective, n_trials=1)
print(f"SQLite    : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (MySQL)
study = optuna.create_study(storage="mysql+pymysql://root:test@localhost/mysql")
try:
    study.optimize(objective, n_trials=1)
except Exception as e:
    print(e)
print(f"MySQL     : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (PostgreSQL)
study = optuna.create_study(storage="postgresql+psycopg2://postgres:test@localhost:15432/postgres")
study.optimize(objective, n_trials=1)
print(f"PostgreSQL: Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# Redis storage
study = optuna.create_study(storage="redis://localhost:6379/0")
try:
    study.optimize(objective, n_trials=1)
except Exception as e:
    print(e)
finally:
    print(f"Redis     : Trial state is {study.trials[-1].state}")
print(study.trials[-1])
```

```
venv) mamu@HideakinoMacBook-puro 3205-3206 % python inf-report.py
[I 2022-02-03 16:39:44,183] A new study created in memory with name: no-name-55b7342a-1401-4c04-a5b0-b3ef6ec14c84
[I 2022-02-03 16:39:44,185] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
In-memory : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[1.0], datetime_start=datetime.datetime(2022, 2, 3, 16, 39, 44, 184687), datetime_complete=datetime.datetime(2022, 2, 3, 16, 39, 44, 184845), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={1: inf}, trial_id=0, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:39:44,274] A new study created in RDB with name: no-name-3ce44212-c256-42d1-914e-ba8ab64bd5a0
[I 2022-02-03 16:39:44,349] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
SQLite    : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[1.0], datetime_start=datetime.datetime(2022, 2, 3, 16, 39, 44, 295707), datetime_complete=datetime.datetime(2022, 2, 3, 16, 39, 44, 320515), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={1: inf}, trial_id=29, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:39:44,482] A new study created in RDB with name: no-name-93b70a4e-4a32-475d-bfbc-ac071b8fbbc4
[I 2022-02-03 16:39:44,706] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
MySQL     : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[1.0], datetime_start=datetime.datetime(2022, 2, 3, 16, 39, 44, 544455), datetime_complete=datetime.datetime(2022, 2, 3, 16, 39, 45), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={1: inf}, trial_id=49, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:39:44,935] A new study created in RDB with name: no-name-7e646d7c-53b3-4e67-98cc-dc2b3c2c194d
[I 2022-02-03 16:39:45,083] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
PostgreSQL: Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[1.0], datetime_start=datetime.datetime(2022, 2, 3, 16, 39, 44, 979182), datetime_complete=datetime.datetime(2022, 2, 3, 16, 39, 45, 19134), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={1: inf}, trial_id=28, state=TrialState.COMPLETE, value=None)
/Users/mamu/Dev/pfn/optuna/optuna/storages/__init__.py:29: ExperimentalWarning: RedisStorage is experimental (supported from v1.4.0). The interface can change in the future.
  return _CachedStorage(RedisStorage(storage))
[I 2022-02-03 16:39:45,100] A new study created in Redis with name: no-name-0000000017
[I 2022-02-03 16:39:45,107] Trial 0 finished with value: 1.0 and parameters: {}. Best is trial 0 with value: 1.0.
Redis     : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[1.0], datetime_start=datetime.datetime(2022, 2, 3, 16, 39, 45, 102660), datetime_complete=datetime.datetime(2022, 2, 3, 16, 39, 45, 104854), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={1: inf}, trial_id=17, state=TrialState.COMPLETE, value=None)
```

</details>

<details>
<summary>Test for study.tell</summary>

```python
import optuna
import numpy as np


# In-memory storage
study = optuna.create_study()
trial = study.ask()
study.tell(trial, float(np.inf))
print(f"In-memory : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (SQLite)
study = optuna.create_study(storage='sqlite:///../sqlite.db')
trial = study.ask()
study.tell(trial, float(np.inf))
print(f"SQLite    : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (MySQL)
study = optuna.create_study(storage="mysql+pymysql://root:test@localhost/mysql")
trial = study.ask()
try:
    study.tell(trial, float(np.inf))
except Exception as e:
    print(e)
print(f"MySQL     : Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# RDB storage (PostgreSQL)
study = optuna.create_study(storage="postgresql+psycopg2://postgres:test@localhost:15432/postgres")
trial = study.ask()
study.tell(trial, float(np.inf))
print(f"PostgreSQL: Trial state is {study.trials[-1].state}")
print(study.trials[-1])

# Redis storage
study = optuna.create_study(storage="redis://localhost:6379/0")
trial = study.ask()
try:
    study.tell(trial, float(np.inf))
except Exception as e:
    print(e)
finally:
    print(f"Redis     : Trial state is {study.trials[-1].state}")
print(study.trials[-1])
```

```
(venv) mamu@HideakinoMacBook-puro 3205-3206 % python inf-tell.py 
[I 2022-02-03 16:42:16,267] A new study created in memory with name: no-name-bfd972f3-6428-4f86-8781-269269a25e0b
In-memory : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 42, 16, 267776), datetime_complete=datetime.datetime(2022, 2, 3, 16, 42, 16, 267909), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=0, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:42:16,336] A new study created in RDB with name: no-name-2919cfc3-61c8-46c6-bc1f-7b5884d36e2c
SQLite    : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 42, 16, 354312), datetime_complete=datetime.datetime(2022, 2, 3, 16, 42, 16, 367098), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=30, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:42:16,511] A new study created in RDB with name: no-name-6e099bcf-cdea-457f-bfb3-2e3f56b6cb9c
MySQL     : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 42, 16, 565982), datetime_complete=datetime.datetime(2022, 2, 3, 16, 42, 17), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=50, state=TrialState.COMPLETE, value=None)
[I 2022-02-03 16:42:16,868] A new study created in RDB with name: no-name-15f997d7-266d-40ec-8d64-51ad95796f27
PostgreSQL: Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 42, 16, 911200), datetime_complete=datetime.datetime(2022, 2, 3, 16, 42, 16, 936047), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=29, state=TrialState.COMPLETE, value=None)
/Users/mamu/Dev/pfn/optuna/optuna/storages/__init__.py:29: ExperimentalWarning: RedisStorage is experimental (supported from v1.4.0). The interface can change in the future.
  return _CachedStorage(RedisStorage(storage))
[I 2022-02-03 16:42:16,974] A new study created in Redis with name: no-name-0000000018
Redis     : Trial state is TrialState.COMPLETE
FrozenTrial(number=0, values=[inf], datetime_start=datetime.datetime(2022, 2, 3, 16, 42, 16, 976742), datetime_complete=datetime.datetime(2022, 2, 3, 16, 42, 16, 978574), params={}, distributions={}, user_attrs={}, system_attrs={}, intermediate_values={}, trial_id=18, state=TrialState.COMPLETE, value=None)
```

</details>

The test of `test_store_infinite_values` was created for only RDB, which automatically converts the value greater than `np.finfo(np.float32).max` into `float("inf")`. Redis storage does not have this specification. Therefore, I guess, @masap skip the test if the `input` is not `+/- float("inf")`.

Comment (HideakiImamura) [reply tests/storages_tests/test_with_server.py:128]: We introduced a specification in #3238 that converts values larger than `np.finfo(np.float32).max` to `float("inf")` in the RDB.

Comment (masap) [reply tests/storages_tests/test_with_server.py:128]: > Therefore, I guess, @masap skip the test if the input is not +/- float("inf").

My code is doing it...

float("inf") -> doing test
-float("inf") -> doint test
other -> skip test

Comment (masap) [reply tests/storages_tests/test_with_server.py:128]: > In addition, how about replacing this silent return with pytest.skip("RedisStorage does not accept float('inf')") ?

I will fix to return "RedisStorage does not convert max/min to inf/-inf".

File: .github/workflows/tests-dbstorage.yml
Comment (masap) [reply .github/workflows/tests-dbstorage.yml:68]: Ok, I will fix it.

File: tests/storages_tests/test_with_server.py
Comment (masap) [reply tests/storages_tests/test_with_server.py:128]: > Therefore, I guess, @masap skip the test if the input is not +/- float("inf").

Yes. (I removed previous my comment because I mis-understood your comment)

> We introduced a specification in https://github.com/optuna/optuna/pull/3238 that converts values larger than np.finfo(np.float32).max to float("inf") in the RDB.

Yes. I will add the link to it.

File: .github/workflows/tests-dbstorage.yml
Comment (toshihikoyanase) [reply .github/workflows/tests-dbstorage.yml:1]: Thank you for your explanation. I overlooked the protection.
In my understanding, a CI file which does not exist in the `optuna/optuna` master will be just ignored. Could you share the error massages of the CI failure if you still have it?

> the first commit log

I'm sorry but I missed [it](https://github.com/optuna/optuna/pull/3258/commits/0748e651c45e7a0fbc2d45c266e48fd4a6c2de5a). If you put it in the description of the PR, the reviewers as well as future developers can  easily access the reason why you keep the file name.

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [reply tests/storages_tests/test_with_server.py:124]: Thank you for your comment. It is related to the storage API reconstruction discussed in https://github.com/optuna/optuna/issues/2943. Please give me some more time as I am discussing this with other committers.

File: .github/workflows/tests-dbstorage.yml
Comment (toshihikoyanase) [reply .github/workflows/tests-dbstorage.yml:68]: Thank you!

**Review by toshihikoyanase — COMMENTED** [review]
Sorry for the late reply. The policy of `get_storage`  not decided yet, but I added some comments on the other points. Please take a look when you have time.

File: tests/storages_tests/test_with_server.py
Comment (masap) [reply tests/storages_tests/test_with_server.py:124]: OK, I will wait for comment.

File: tests/storages_tests/test_with_server.py
Comment (masap) [reply tests/storages_tests/test_with_server.py]: Fixed.

File: .github/workflows/tests-dbstorage.yml
Comment (masap) [reply .github/workflows/tests-dbstorage.yml]: Fixed.

File: .github/workflows/tests-dbstorage.yml
Comment (masap) [reply .github/workflows/tests-dbstorage.yml]: > Could you share the error massages of the CI failure if you still have it?

I made a PR #3300 that shows effect of renaming the job name.
RDB Storage tests itself already finished but tests-rdbstorage (3.x) still waits for finish of old name jobs.

File: tests/storages_tests/test_with_server.py
Comment (HideakiImamura) [reply tests/storages_tests/test_with_server.py:124]: @masap We discussed whether or not to make `_CachedStorage` available for users. The conclusion was that it was too early to release it yet, so we decided not to release it in V3. Related to that, we also decided not to add a "raw" option to `get_storage`.

Comment (masap) [reply tests/storages_tests/test_with_server.py]: Ok, I removed related comments from `test_store_infinite_values` and `test_fail_stale_trials_raw`.

File: .github/workflows/tests-dbstorage.yml
Comment (toshihikoyanase) [reply .github/workflows/tests-dbstorage.yml]: > RDB Storage tests itself already finished but tests-rdbstorage (3.x) still waits for finish of old name jobs.

I see. `test-rdbstorage` is makred as required, and the corresponding jobs could not be found after renaming the job name. Thank you for your explanation.

So, it's impossible to rename it with this PR. Then, could you add a TODO comment to rename the job name?

Comment (masap) [reply .github/workflows/tests-dbstorage.yml]: > could you add a TODO comment to rename the job name?

Fixed.

**Comment by masap** [comment]
Unexpectedly `Tests / tests (3.9) (pull_request)` fails without any error message, could anybody re-run this ?

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update! LGTM.

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [inline tests/storages_tests/test_with_server.py:12]: A nit, but `._redis` can be removed.
```suggestion
from optuna.storages import RedisStorage
```

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [inline tests/storages_tests/test_with_server.py:123]: A nit-picky, but could you write the URL instead of the PR/Issue number?
```suggestion
        # Only RDB can convert max/min value to inf. It was introduced by 
        # https://github.com/optuna/optuna/pull/3238.
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your update. The change basically looks good to me, but I suggested two small updates.

File: .github/workflows/tests-dbstorage.yml
Comment (toshihikoyanase) [reply .github/workflows/tests-dbstorage.yml]: Thanks!

File: .github/workflows/tests-dbstorage.yml
Comment (toshihikoyanase) [reply .github/workflows/tests-dbstorage.yml]: Thank you!

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [reply tests/storages_tests/test_with_server.py]: Thanks!

File: tests/storages_tests/test_with_server.py
Comment (masap) [reply tests/storages_tests/test_with_server.py]: Fixed also `tests/storages_tests/test_storages.py`.

File: tests/storages_tests/test_with_server.py
Comment (masap) [reply tests/storages_tests/test_with_server.py]: Fixed.

**Comment by masap** [comment]
Fixed conflict caused by commit ac00edece6eadced3bb39e0c5709fb77d3c23f34.

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [reply tests/storages_tests/test_with_server.py]: Thanks!

File: tests/storages_tests/test_with_server.py
Comment (toshihikoyanase) [reply tests/storages_tests/test_with_server.py]: Thank you!

**Pull request merged**
